### PR TITLE
Fix vSphere firewalling

### PR DESCRIPTION
**Pull request by axw**
## Description of change

This PR fixes the regression in functionality of the vsphere instance firewalling code. We disable strict host checking in the SSH connections from the controller to machine hosts made for iptables management. We also fix iptables parsing and rule creation/deletion command rendering. Abstractions around iptables have been moved to a new package, network/iptables.

## QA steps

1. juju bootstrap vsphere --config external-network="<external-network>"
2. juju deploy ubuntu
3. juju run --unit ubuntu -- open-port 1234
(check on the ubuntu/0 unit's machine that no iptables rules are added)
4. juju expose ubuntu
(check on ubuntu/0's machine that an iptables rule is added)
5. juju ssh -m controller 0 "sudo service jujud-machine-0 restart"
(check that no duplicate rules are added)
6. juju unexpose ubuntu
(check on ubuntu/0's machine that the iptables rule is removed)

## Documentation changes

None.

## Bug reference

Partially fixes https://bugs.launchpad.net/juju/+bug/1732665

### CODE REVIEW

**Comment by axw** [comment]
[vsphere firewalling](https://trello.com/c/tHOWErCf/70-vsphere-firewalling)

File: provider/common/instance_configurator.go [hunk 1]
```diff
@@ -48,24 +46,48 @@ type sshInstanceConfigurator struct {
 func NewSshInstanceConfigurator(host string) InstanceConfigurator {
 	options := ssh.Options{}
 	options.SetIdentities("/var/lib/juju/system-identity")
+
+	// Disable host key checking. We're sending any sensitive data
```
Comment (wallyworld) [inline provider/common/instance_configurator.go]: not sending?

Comment (axw) [reply provider/common/instance_configurator.go]: heh thanks, fixed

**Comment by axw** [comment]
$$merge$$

**Pull request merged**